src/cast/safe/size.rs

```diff
@@ -0,0 +1,103 @@
+/// Cast between different sized numbers without changing sign.
+///
+/// The `SizeFrom<T>` trait is similar to `std::convert::From<T>`. However,
+/// while `std::convert::From<T>` performs a logical conversion, `SizeFrom<T>`
+/// performs a bitwise cast from a number to a number of the same sign but
+/// of a possibly different size. `TrimFrom<T>` will **never** change from an
+/// integer of one signedness to the other.
+///
+/// Unless you really know what you are doing, you probably don't want this
+/// trait. Instead, you should check out the following traits:
+///
+///   * `GrowFrom<T>`
+///   * `TrimFrom<T>`
+///   * `SignCast`
+pub trait SizeFrom<T> {
+    fn size(value: T) -> Self;
+}
```

Comment:
Like my comments on `GrowFrom`, I don't see the value of maintaining an unknown signedness.  If you do know the sign you're dealing with, then constraints on `AsPrimitive` and `Signed`/`Unsigned` should do the same job.

Reply:
See my comment above about readability. When someone reads code and sees `as` they don't know what actually happened. This is about making all behaviors of the cast as explicit as possible.